src/app/pages/system/email/email.component.ts

```diff
@@ -1,366 +1,238 @@
-  client_secret?: string;
-  refresh_token?: string;
+enum SendMethod {
+  Smtp = 'smtp',
+  Gmail = 'gmail',
 }
 
 @UntilDestroy()
 @Component({
-  selector: 'app-email',
-  template: '<entity-form [conf]="this"></entity-form>',
+  templateUrl: './email.component.html',
+  styleUrls: ['./email.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class EmailComponent implements FormConfiguration {
-  queryCall = 'mail.config' as const;
-  updateCall = 'mail.update' as const;
-  entityEdit: EntityFormComponent;
-  rootEmail: string;
-  private oauthCreds$: BehaviorSubject<OAuthData> = new BehaviorSubject({});
-  customSubmit = this.saveConfigSubmit;
-  customActions = [{
-    id: 'send_mail',
-    name: this.translate.instant('Send Test Mail'),
-    function: () => {
-      if (this.rootEmail) {
-        const value = _.cloneDeep(this.entityEdit.formGroup.value);
-        if (!value.send_mail_method) {
-          delete value.pass;
-          value.smtp = false;
-          value.oauth = this.oauthCreds$.getValue();
-        } else {
-          delete value.oauth;
-        }
-        delete value.send_mail_method;
-        delete value.oauth_applied;
-
-        const productType = window.localStorage.getItem('product_type') as ProductType;
-        const mailObj = {
-          subject: 'TrueNAS Test Message',
-          text: `This is a test message from TrueNAS ${productType}.`,
-        };
-        this.ws.call('system.info').pipe(untilDestroyed(this)).subscribe((systemInfo) => {
-          value.pass = value.pass || this.entityEdit.data.pass;
-
-          mailObj['subject'] += ' hostname: ' + systemInfo.hostname;
-          this.dialogRef = this.dialog.open(EntityJobComponent, { data: { title: this.translate.instant('EMAIL') }, disableClose: true });
-          this.dialogRef.componentInstance.setCall('mail.send', [mailObj, value]);
-          this.dialogRef.componentInstance.submit();
-          this.dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
-            this.dialogRef.close(false);
-            this.dialogservice.info(this.translate.instant('Email'), this.translate.instant('Test email sent!'), '500px', 'info');
-          });
-          this.dialogRef.componentInstance.failure.pipe(untilDestroyed(this)).subscribe((job) => {
-            this.dialogRef.componentInstance.setDescription(job.error);
-          });
-        });
-      } else {
-        this.dialogservice.info(this.translate.instant('email'), this.translate.instant('Configure the root user email address.'));
-      }
+export class EmailComponent implements OnInit {
+  sendMethodControl = new FormControl(SendMethod.Smtp);
+
+  form = this.formBuilder.group({
+    fromemail: ['', [Validators.required, Validators.email]],
+    fromname: [''],
+    outgoingserver: [''],
+    port: [null as number, [Validators.required, portRangeValidator()]],
+    security: [MailSecurity.Plain],
+    smtp: [false],
+    user: [''],
```

Comment:
user: ['', Validators.required]

Also please add `[required]="true"` in template.

![image](https://user-images.githubusercontent.com/22980553/163849203-93c87101-94bf-4eaa-8251-1d5c32881047.png)